### PR TITLE
feat: add default `onCanvas` constraints for all shapes

### PR DESCRIPTION
**Pull request by wodeni**
# Description

Related issue/PR: #582 #361 

__Need a generic bboxFromShape function for this feature to work for all shapes.__ Depends on #696 

`maxSize` doesn’t have a clear meaning and this PR replaces them with a default constraint `onCanvas`. All shapes now have a default property `ensureOnCanvas: true`, which Style writers can opt-out by writing, eg `x.icon = Circle { ensureOnCanvas: false }`. Whenever `ensureOnCanvas: true`, the Style compiler generates a onCanvas constraint for that shape. 

In addition, `minSize` is also too vague. This PR changes all examples to have more specific constraints like `ensure lessThan(50, x.icon.r)`

# Implementation strategy and design decisions

* Delete `maxSize` and `minSize`
* Add `onCanvas` constraint in the constraint dictionary
* Add default constraints based on the value of `ensureOnCanvas` in the Style compiler
* Port all examples

# Examples with steps to reproduce them

https://e1aaa942.penrose-panes.pages.dev/gist/df0cc75e79c863ab2a53dbf0a4ffecfd

on-canvas-demo

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally using `yarn test`
- [x] I ran `yarn docs` and there were no errors when generating the HTML site
- [x] My code follows the style guidelines of this project (e.g.: no ESLint warnings)

# Open questions

TODO

### CODE REVIEW

**Comment by wodeni** [comment]
Also just figured out how to use the Cloudflare preview for a quick demo of a PR branch. Here are the steps I took for this example: https://e1aaa942.penrose-panes.pages.dev/gist/df0cc75e79c863ab2a53dbf0a4ffecfd.

* Go to https://panes.penrose.ink, log in, and create a diagram. Doesn't matter if it works on thie deployed site.
* Hit publish and copy the hash after `panes.penrose.ink/gist/`.
* Go to the preview site, type `/gist/<hash>` after the URL. Voilà!

**Comment by wodeni** [comment]
@samestep is working on #698. Once it's done, I think it's okay to merge this PR.

**Comment by wodeni** [comment]
@joshsunshine proposed to change the property to `ensureOnCanvas`, which nicely coincides with the keyword `ensure`.

**Review by samestep — APPROVED** [review]
Looks good to me! I don't know enough about the Style compiler etc to review most of this, but it looks like it'll interface correctly with what I did in #698.

**Comment by wodeni** [comment]
Now that I'm using the generic bbox functions implemented in #696, the overall tests started to time out in CI. This is the same performance issue mentioned in #652, likely because of the increased complexity and number of constraints due to `onCanvas`. I won't merge this PR until #652 is resolved.

File: examples/set-theory-domain/venn.sty
Comment (jiriminarcik) [inline examples/set-theory-domain/venn.sty:16]: delete

File: packages/core/src/contrib/Constraints.ts
Comment (jiriminarcik) [inline packages/core/src/contrib/Constraints.ts:161]: Not sure whether this is true, but `mul(canvasWidth, constOf(0.5))` may be a bit more efficient?

**Review by jiriminarcik — APPROVED** [review]
Looks good to me

**Pull request merged**
